examples/basic/hello-world-events.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * This example showcases how users can listen to various events exposed from golem-js
+ */
+import { GolemNetwork } from "@golem-sdk/golem-js";
+import { pinoPrettyLogger } from "@golem-sdk/pino-logger";
+
+(async () => {
+  const glm = new GolemNetwork({
+    logger: pinoPrettyLogger({
+      level: "info",
+    }),
+    payment: {
+      driver: "erc20",
+      network: "holesky",
+    },
+  });
+
+  try {
+    await glm.connect();
+
+    glm.market.events.on("agreementConfirmed", (agreement) => {
+      console.log("Agreement '%s' confirmed", agreement.id);
+    });
+
+    glm.market.events.on("agreementTerminated", (agreement, terminatedBy, reason) => {
```

Comment:
Add a few more events and make the example depict thee lifecycle in its general form. Also, Switch to passing the whole `event` object so that the listener functions can be simplified.